guides/developer/plugin-creators/extensionpoint-plugin-walkthrough.md

```diff
@@ -0,0 +1,355 @@
+Items in this list must be from the [shared libraries](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/plugins/sharedLibraries.ts#L32) exposed to plugin creators.
+
+### Dependencies
+As mentioned above, Spinnaker exposes libraries for plugins to use. Define dependencies in package.json. For this example plugin, the dependencies are:
+
+```
+"@spinnaker/core": "0.0.432",
+"react": "^16.12.0",
+"react-dom": "^16.12.0"
+```
+
+## Writing The Frontend
+
+```
+import * as React from 'react';
+import { IStageTypeConfig, IStageConfigProps } from '@spinnaker/core';
+
+const customStage: IStageTypeConfig = {
+  label: 'Random Wait',
+  description: 'Stage that waits a random amount of time up to the max inputted',
+  key: 'randomWait',
+  component: RandomWaitStage,
+};
+
+function setMaxWaitTime(event: React.SyntheticEvent, props: IStageConfigProps) {
+  let target = event.target as HTMLInputElement;
+  props.updateStageField({'maxWaitTime': target.value});
+}
+
+// Our stage component
+function RandomWaitStage(props: IStageConfigProps) {
+  return (
+    <div>
+      <label>
+          Max Time To Wait
+          <input value={props.stage.maxWaitTime} onChange={(e) => setMaxWaitTime(e, props)} id="maxWaitTime" />
+      </label>
+    </div>
+  );
+}
+
+const plugin = {
+  name: 'randomWait',
+  stages: [customStage],
+};
+
+export { plugin };
+```
+
+### IStageTypeConfig
+Define Spinnaker Stages with IStageTypeConfig. Required [options:](https://github.com/spinnaker/deck/blob/abac63ce5c88b809fcf5ed1509136fe96489a051/app/scripts/modules/core/src/domain/IStageTypeConfig.ts)
+- label -> The name of the Stage
+- description -> Long form that describes what the Stage actually does
+- key -> A unique name for the Stage
+- component -> The rendered React component
+
+### IStageConfigProps
+`IStageConfigProps` defines properties passed to all Spinnaker Stages. See [IStageConfigProps.ts](https://github.com/spinnaker/deck/blob/master/app/scripts/modules/core/src/pipeline/config/stages/common/IStageConfigProps.ts) for a complete list of properties. Pass a JSON object to the `updateStageField` method to add the `maxWaitTime` to the Stage.
+
+### RandomWaitStage
+This method returns [JSX](https://reactjs.org/docs/introducing-jsx.html) that gets displayed to the plugin user.
+
+### How Spinnaker Loads The Plugin
+Each plugin must export an object named `plugin`. You can only add Stages to this object. At startup, Spinnaker looks at `plugin.stages` and adds each defined Stage to the Stage Registry.
+
+# Writing The Plugin Manifest
+
+Here is an example of a plugin manifest:
+```
+name: armory/randomWaitStage
+description: Copies S3 files to different locations
+manifestVersion: plugins/v1
+version: 1.2.3
+options:
+  s3:
+    username: user
+    password: pass
+resources:
+  orca:
+  - https://stage-plugin-test.s3-us-west-2.amazonaws.com/stage-plugin-0.0.1-SNAPSHOT.jar
+```
```

Comment:
Is this a Halyard config format? I've not seen anything that resembles this before.

Reply:
Don't know where Brandon go this - removed along with the entire "frontend" section, which was outdated.